### DISCUSSION TITLE
Slow table performance / huge number of Blade Components.

### DISCUSSION THREAD

**Discussion by bennettblack** [filamentphp/filament#5098] [Help] [answered]
Hello,

I've installed the Filament Admin Panel into a fresh Laravel app. It seems that the 'list' view in the admin panel for resources is very slow. One of my resources has 100K+ records, the other has only 200 records and they both seem to "hang" when changing the current page of results. I noticed in the debugbar that there are a LOT of view components being rendered (1256) on a page with only 50 results. Is that normal? Could that be contributing to the "laggy" behavior?

![image](https://user-images.githubusercontent.com/51721783/204638934-4c079fc6-2b94-415c-9a60-74cbd29d3f1a.png)

I am running Laravel 9.x + PHP 8.1 locally with Windows 10.

**Comment by danharrin** [comment]
Hey!

- 100k records is likely not contributing to the time it takes for the table to render. That would only affect database query speed, not rendering speed.
- 1256 views are not going to be affecting the speed much. Think of each view as just being a simple PHP `include`. If we didn't extensively use Blade components, the Filament codebase would be unmaintainable as duplicate HTML would be absolutely everywhere.
- Rendering 4 columns for 50 records on my machine (stock M1 Macbook Pro) takes 115ms / 7MB memory. It's even faster in production (basic $6 DigitalOcean server).

The memory usage looks very very high, which is a symptom of something else. I have some questions:
- Do you have opcache enabled?
- Do you have xdebug enabled?
- Which EXACT Laravel version are you using?
- What is the speed & memory usage for a non-Filament page in your app?

**Reply by bennettblack** [reply]
Hi Dan,

Here's my specifics:

1. Laravel 9.42.0
2. OP Cache Disabled (for dev environment)
3. Xdebug disabled

For non-filament pages in my dev environment I'm running about 1.5seconds / 30MB ram.

I decided to test this in production (Ubuntu 22.x, 8GB Ram, 2 cores, OPCache enabled) and the numbers are much better:
![image](https://user-images.githubusercontent.com/51721783/204892011-9a8972c0-16d3-4002-80e5-cf23d79fea9f.png)

Im curious - what's your environment like on your mac and what browser are you using? I have an M2 air I could test this on.

**Reply by danharrin** [reply]
1.5seconds / 30MB ram outside of Filament is still huge... this is sub 250ms on 95% of machines. Which indicates that your environment is generally slow, and the memory usage is huge? Are you doing lots of work in service providers / middleware?

The production test numbers look much more realistic but still a little slow

Have you tested Filament / Laravel on a fresh project?

I'm using Valet & Safari

**Reply by bennettblack** [reply]
Nothing abnormal with providers or middleware - this is a pretty fresh project. Using @underdpt 's suggestion to disable the debugbar's "Views" tab did help a lot.

I did a comparison with this and one other Laravel project on my Windows dev environment VS my M2 Mac environment (using Microsoft Edge + Valet) and it seems that projects use substantially less RAM on my Mac. Not sure what to attribute the difference to. 

Thanks for the help troubleshooting!

**Comment by underdpt** [comment, marked as answer]
Hello,

I've run into this also. This is caused by the DebugBar `views` collector. If you don't need it, you can disable it on your `config/debugbar.php` file:

```php
    'collectors' => [
       ...
       'views'           => false,  // Views with their data
       ...
   ]
```
